src/binding/defaultBindings/style.js

```diff
@@ -12,15 +12,30 @@ ko.bindingHandlers['style'] = {
             if (jQueryInstance) {
                 jQueryInstance(element)['css'](styleName, styleValue);
             } else {
-                styleName = styleName.replace(/-(\w)/g, function (all, letter) {
-                    return letter.toUpperCase();
-                });
+                var previousStyle;
+                var postStyle;
+                var requiresGetPropertyValue = (styleName.substring(0, 2) === "--");
```

Comment:
Perhaps instead `var isCustomPropertyName = /^--/.test(styleName);`
And possibly declaring the regexp in some outer scope to allow reuse.